plugins/modules/external_usergroup.py

```diff
@@ -97,7 +107,15 @@ def main():
                 entity = external_usergroup
 
         module.set_entity('entity', entity)
-        module.auto_lookup_entities()
+
+        auth_source = module.lookup_entity('auth_source')
+        if auth_source.get('type') == 'AuthSourceExternal':
+            module.set_entity('auth_source_external', auth_source)
+        elif auth_source.get('type') == 'AuthSourceLdap':
+            module.set_entity('auth_source_ldap', auth_source)
+        else:
+            module.fail_json(msg="Unsupported authentication source type: {0}".format(auth_source.get('type')))
+
         module.ensure_entity('external_usergroups', module.foreman_params, entity, params)
```

Comment:
I know, this is unrelated, but should this be `module.run()`? Maybe in another PR.
The lines above it look very clean now.

Reply:
`run(params=params)`, but yes.